### PR TITLE
dwarfs: 0.7.5 → 0.9.9

### LINKED ISSUE #306563: Update request: dwarfs 0.7.5 → 0.9.8
- Package name: dwarfs
- Latest released version: 0.9.8

- Current version on the unstable channel: 0.7.5
- Current version on the stable/release channel: 0.7.2

- [x] Checked the [nixpkgs pull requests](https://github.com/NixOS/nixpkgs/pulls)

**Notify maintainers**
@keksbg

-----

Note for maintainers: Please tag this issue in your PR.

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

### PR DESCRIPTION
**Pull request by luftmensch-luftmensch**
## Description of changes
Closes  #306563

## Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [X] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by luftmensch-luftmensch** [comment]
```
> nix build .#dwarfs
> tree result/
 result
├──  bin
│  ├──  dwarfs
│  ├──  dwarfsbench
│  ├──  dwarfsck
│  ├──  dwarfsextract
│  ├──  mkdwarfs
│  └──  mount.dwarfs -> dwarfs
├──  sbin -> bin
└──  share
   └──  man
      ├──  man1
      │  ├──  dwarfs.1.gz
      │  ├──  dwarfsck.1.gz
      │  ├──  dwarfsextract.1.gz
      │  └──  mkdwarfs.1.gz
      └──  man5
         └──  dwarfs-format.5.gz
```

```
./result/bin/dwarfs --help                                                                                                                                       0.015s 
     ___                  ___ ___
    |   \__ __ ____ _ _ _| __/ __|         Deduplicating Warp-speed
    | |) \ V  V / _` | '_| _|\__ \      Advanced Read-only File System
    |___/ \_/\_/\__,_|_| |_| |___/         by Marcus Holland-Moritz

dwarfs (v0.9.9, fuse version 35)
built on x86_64, Linux-6.1.90, GNU 13.2.0

usage: dwarfs <image> <mountpoint> [options]

DWARFS options:
    -o cachesize=SIZE      set size of block cache (512M)
    -o workers=NUM         number of worker threads (2)
    -o mlock=NAME          mlock mode: (none), try, must
    -o decratio=NUM        ratio for full decompression (0.8)
    -o offset=NUM|auto     filesystem image offset in bytes (0)
    -o enable_nlink        show correct hardlink numbers
    -o readonly            show read-only file system
    -o (no_)cache_image    (don't) keep image in kernel cache
    -o (no_)cache_files    (don't) keep files in kernel cache
    -o debuglevel=NAME     error, warn, info, debug, trace
    -o tidy_strategy=NAME  (none)|time|swap
    -o tidy_interval=TIME  interval for cache tidying (5m)
    -o tidy_max_age=TIME   tidy blocks after this time (10m)
    -o perfmon=name[,...]  enable performance monitor

FUSE options:
    -h   --help            print help
    -V   --version         print version
    -d   -o debug          enable debug output (implies -f)
    -f                     foreground operation
    -s                     disable multi-threaded operation
    -o clone_fd            use separate fuse device fd for each thread
                           (may improve performance)
    -o max_idle_threads    the maximum number of idle worker threads
                           allowed (default: -1)
    -o max_threads         the maximum number of worker threads
                           allowed (default: 10)
```

**Comment by Misterio77** [comment]
Result of `nixpkgs-review pr 312656` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>dwarfs</li>
  </ul>
</details>

File: pkgs/tools/filesystems/dwarfs/default.nix [hunk 1]
```diff
@@ -1,50 +1,54 @@
...
-
-stdenv.mkDerivation rec {
+let
   pname = "dwarfs";
-  version = "0.7.5";
-
+  version = "0.9.9";
+in
+stdenv.mkDerivation {
+  inherit pname version;
```
Comment (SuperSandro2000) [inline pkgs/tools/filesystems/dwarfs/default.nix:31]: Avoiding rec here is kinda pointless but whatever

**Review by SuperSandro2000 — APPROVED** [review]

File: pkgs/tools/filesystems/dwarfs/default.nix [hunk 2]
```diff
@@ -1,50 +1,54 @@
...
   src = fetchFromGitHub {
     owner = "mhx";
     repo = "dwarfs";
-    rev = "v${version}";
+    rev = "refs/tags/v${version}";
     fetchSubmodules = true;
     hash = "sha256-Zzm2SaFR31TBBMDfgJulVbqsJBh1He2wBFzHRC/c5vg=";
```
Comment (nh2) [inline pkgs/tools/filesystems/dwarfs/default.nix:37]: The `version` was bumped but the `hash` is unchanged.

So this is really building the old version!

**Pull request merged**